pkgs/applications/editors/vscode-with-extensions/default.nix

```diff
@@ -0,0 +1,107 @@
+/*
+  `vsixExtensions`
+   :  A set of vscode extensions to be installed alongside the editor. Here's a an
+      example:
+
+      ~~~
+      vscode-with-extensions.override {
+        vsixExtensions = {
+          "nix" = {
+            publisher = "bbenoist";
+            version = "1.0.1";
+            sha256 = "0zd0n9f5z1f0ckzfjr38xw2zzmcxg1gjrava7yahg5cvdcw6l35b";
+          };
```

Comment:
Seems like sensible suggestions. I assume you meant that `vsixExtensions` would take a list of derivations instead of the attr set of sources. If so, I would then rename this to `vscodeExtensions` at this would no longer be simple descriptions how to retrieves `*.vsix` files but full fledged derivations with the `*.vsix` in its already installed / unpacked form.

The `vscode-extensions` seems a great idea in later iterations. I'm not quite sure how to handle the licensing of the extensions so as a first iteration I would rather provide simple tools so that users can concisely provide their own listing of extension from the marketplace directly at this point, overriding the derivation and as you propose combine these with full fledged derivations if need be. I was thinking of a function taking a list of sources (same as the attrset above but the name moved to items and the attrset -> list named something like `extensionsFromVscodeMarketplace`) and returning a list of derivation. The function could be implemented using a `fetchVsixFromVscodeMarketplace` (analogous to the `fetchFromGitHub` used by `buildVimPluginFrom2Nix`).

In this first iteration, I'm not quite there managing dependencies same as `buildVimPluginFrom2Nix`. Could this part be done in later iterations?

This first draft would be used like this:

~~~{.nix}
(vscode-with-extensions.override {
  vscodeExtensions = extensionsFromVscodeMarketplace [
    { 
      name = "nix"
      publisher = "bbenoist";
      version = "1.0.1";
      sha256 = "0zd0n9f5z1f0ckzfjr38xw2zzmcxg1gjrava7yahg5cvdcw6l35b";
    }
  ]
  # ++ with vscode-extensions; [  drvFromThisNamespaceCouldBeAddedAtSomePoint ] */ 
  # ++ moreComplexDerivationWithPatchesOrFromOtherSourcesConcatenedHere*/;
})
~~~

The helpers could be provided in a `vscodeUtils` namespace same as vim.

Does it seem ok for you?